OrbitGl/DataView.h

```diff
@@ -47,6 +52,7 @@ class DataView {
   virtual size_t GetNumElements() { return indices_.size(); }
   virtual std::string GetValue(int /*a_Row*/, int /*a_Column*/) { return ""; }
   virtual std::string GetToolTip(int /*a_Row*/, int /*a_Column*/) { return ""; }
+  bool update_selection() { return update_selection_; }
```

Comment:
This "update_selection()/SetUpdateSelectedIndices" code is unclear. The naming should be consistent. Also, the mechanism seems to be there to prevent certain operations when triggered from the Refresh() method. Why? What is the issue exactly? This seems a bit hacky, if we can't find an alternative, the existing code should be clearly documented to explain the issue.

Reply:
Thanks for pointing out this. To make it easier understood and meanwhile keep a consistent naming, I changed naming as: `update_selection_` ->`is_filtering_or_sorting_`; `SetUpdateSelectedIndices(...)`->`SetIsFilteringOrSorting(...)`.

The `is_filtering_or_sorting_` is used to determine whether the user's selection information (e.g., `SamplingDataView::selected_indices_` and `LiveFunctionsDataView::selected_function_address_`) should be changed.
More specifically, if the calling of `DataView::OnSelect` or `DataView::OnMultiSelect` is caused by filtering or sorting, then `is_filtering_or_sorting_ == true`, we will not change the selection information in the dataview and so the dataview can remember user's last selection during filtering or sorting.

The mechanism does not prevent operations in `Refresh()`. When `Refresh()` is triggered for the **Live view**:
- In previous version, only `layoutAboutToBeChanged()` and `layoutChanged()` are emitted and the following re-selection operations are skipped;
- In current version, if the `Refresh()` is triggered by `OnFilter` or `OnSort`, with this mechanism, we are able to continue the following re-selection operations for the Live view. Therefore, when filtering or sorting, we can update the selection in the Live view to be the same as "previous selected function" rather than "previous selected row".